### PR TITLE
feat: support more python versions

### PR DESCRIPTION
**Pull request by Barabazs**
## Summary

This PR allows for using Aineko on different python version: 3.8, 3.9, 3.10 & 3.11

## Motivation

Make Aineko more accessible by being less restrictive for the version.

## Author Checklist

- [x] Changes are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Documentation is updated if necessary.
- [x] Status checks pass.
- [x] If the version number is changed, it is updated in `pyproject.toml` , `aineko/__init__.py`, `aineko/templates/first_aineko_pipeline/{{cookiecutter.project_slug}}/pyproject.toml`.

## Reviewer Checklist

- [ ] Title is accurate and formatted appropriately.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Docs are updated if necessary.
- [ ] Code is pythonic and consistent with the rest of the codebase.
- [ ] The code is consistent with the [style guide](https://google.github.io/styleguide/pyguide.html).

### CODE REVIEW

**Comment by craycrayfish** [comment]
The complete combination of checks should not be done for every commit - let's trigger the version checks when we're going to release a new version. A potential criteria could be checking the branch name to be `chore/bump-aineko-version` etc, but we would need to set some standard there. Alternatively, we could do some diff checking for the version defined in pyproject.toml.

**Comment by Barabazs** [comment]
> The complete combination of checks should not be done for every commit - let's trigger the version checks when we're going to release a new version. A potential criteria could be checking the branch name to be `chore/bump-aineko-version` etc, but we would need to set some standard there. Alternatively, we could do some diff checking for the version defined in pyproject.toml.

Why would you want to do that? 

I'm in favor of doing it for every push because:
- The total time spent on testing is identical to before since the tests run in parallel.
- Merging incompatible code and cleaning up later is inefficient.

**Comment by craycrayfish** [comment]
> > The complete combination of checks should not be done for every commit - let's trigger the version checks when we're going to release a new version. A potential criteria could be checking the branch name to be `chore/bump-aineko-version` etc, but we would need to set some standard there. Alternatively, we could do some diff checking for the version defined in pyproject.toml.
> 
> Why would you want to do that?
> 
> I'm in favor of doing it for every push because:
> 
> * The total time spent on testing is identical to before since the tests run in parallel.
> * Merging incompatible code and cleaning up later is inefficient.

This would increase the CircleCI cost by 5x for those jobs. Maybe not an issue yet since the total costs aren't very high. I'm fine with merging as is and then figuring out how to optimize later when it becomes a problem.

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,19 +16,42 @@ workflows:
...
+                - "3.9"
+                - "3.10"
+                - "3.11"
+          python_version: << matrix.python_version >>
           filters:
             branches:
               ignore: develop
           prj_dir: aineko
-          python_version: << pipeline.parameters.python_version >>
+          # python_version: << matrix.python_version >>
```
Comment (craycrayfish) [inline .circleci/config.yml]: Can we remove the commented lines?

**Review by craycrayfish — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (Barabazs) [reply .circleci/config.yml]: Addressed in 37d7433a8df88bc56f2ebfe921d5753191d1d321

**Pull request merged**